### PR TITLE
Add 'query' filter support.

### PR DESCRIPTION
**Pull request by krystian8207**
The new filter allows to define complex filtering queries with the widget from <https://github.com/r-world-devs/shinyQueryBuilder> package.

### CODE REVIEW

File: examples/bs4.R [hunk 1]
```diff
@@ -1,6 +1,6 @@
 library(magrittr)
 library(cohortBuilder)
-library(shinyCohortBuilder)
+pkgload::load_all()
```
Comment (galachad) [inline examples/bs4.R]:
```suggestion
library(shinyCohortBuilder)
```

We should keep `library(shinyCohortBuilder)` in the examples.

File: examples/query_filter.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
+library(magrittr)
+library(cohortBuilder)
+pkgload::load_all()
```
Comment (galachad) [inline examples/query_filter.R]:
```suggestion
library(shinyCohortBuilder)
```

Use library in the examples.

File: renv.lock [hunk 3]
```diff
@@ -1048,6 +1054,22 @@
       ],
       "Hash": "1cba04a4e9414bdefc9dcaa99649a8dc"
     },
+    "queryBuilder": {
+      "Package": "queryBuilder",
+      "Version": "0.1.0",
+      "Source": "Local",
+      "RemoteType": "local",
+      "RemoteUrl": "../queryBuilder",
```
Comment (galachad) [inline renv.lock]: It's not necessary, but to simplify collaboration, we should keep records from github not local filesystem.

File: renv.lock [hunk 4]
```diff
@@ -1250,11 +1272,30 @@
       ],
       "Hash": "0be068d8598aca4ee61cc19f03ef51bf"
     },
+    "shinyQueryBuilder": {
+      "Package": "shinyQueryBuilder",
+      "Version": "0.1.0",
+      "Source": "Local",
+      "RemoteType": "local",
+      "RemoteUrl": "../shinyQueryBuilder",
```
Comment (galachad) [inline renv.lock]: The same as for queryBuilder.

File: DESCRIPTION [hunk 5]
```diff
@@ -40,14 +40,15 @@ Imports:
...
-    cohortBuilder (>= 0.2.0),
+    cohortBuilder (>= 0.2.2),
     tryCatchLog,
     highr,
     shinyGizmo (>= 0.4.2),
     rlang (>= 1.0),
     tibble,
     lifecycle
 Suggests:
+    shinyQueryBuilder(>= 0.1.0)
```
Comment (galachad) [inline DESCRIPTION:51]: As we have `shinyQueryBuilder` we should have the check if the package exists in `.gui_filter.query`. We should display the message that the package is required to use a query type filter.

**Review by galachad — APPROVED** [review]
LGTM, looks like all works as expected, few small suggestions.

File: examples/bs4.R (hunk 1 above)
Comment (krystian8207) [reply examples/bs4.R]: done

File: examples/query_filter.R (hunk 2 above)
Comment (krystian8207) [reply examples/query_filter.R]: done

File: renv.lock (hunk 3 above)
Comment (krystian8207) [reply renv.lock]: done

File: renv.lock (hunk 4 above)
Comment (krystian8207) [reply renv.lock]: done

**Pull request merged**